unthink-stack/src/server/framework/service-result.ts

```diff
@@ -0,0 +1,55 @@
+import { ServiceError } from './service-error';
+
+
+enum ServiceResultState {
+  OK,
+  ERROR,
+  NOT_FOUND
+}
+
+export class ServiceResult<ValueModel = unknown, ErrorModel = ServiceError> {
+
+  private readonly state: ServiceResultState;
+
+  public value?: ValueModel;
+  public error?: ErrorModel;
```

Comment:
```suggestion
  public readonly value?: ValueModel;
  public readonly error?: ErrorModel;
```

I don't think this is necessary, but if these are only supposed to be set via the constructor we can make them `readonly`.